### ISSUE TITLE
Failed to load graph correctly when running tests in parallel

### ISSUE THREAD

**Issue by carlos0202** [ent/ent#2623]
When running tests in parallel that create a database, schema, and all from scratch, entc package yields the following error (actual package source names replaced by ***):
```
entc/load: .entc/entc_***_1654531588.go:7:2: cannot find package\n.
entc/entc_***_1654531588.go:8:2: cannot find package\n.
entc/entc_***_654531588.go:9:2: cannot find package\n.
entc/entc_***_1654531588.go:10:2: cannot find package\n.
entc/entc_***_1654531588.go:11:2: cannot find package\n.
entc/entc_***_1654531588.go:16:2: cannot find package
``` 

- [ ] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

Schema load and migration tests runs flawlessly if I'm not using any threading mechanism, but when adding `t.Parallel()` it starts breaking with the error mentioned before.

## Expected Behavior 🤔

All tests loading the same schema should run in parallel in the same way they run when doing sequentially.
## Steps to Reproduce 🕹

Steps:

1. Create some tests that create their own server instance and database while also loading the same schema for different set of tests.
2. Run those tests in parallel using provided go testing helper functions (`t.Parallel()`

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.18  |
| Ent         | 0.10.2   |
| Database    | PostgreSQL   |

**Comment by carlos0202** [comment]
> Hey @carlos0202 and thanks for reporting this issue 👋
> 
> Can you please elaborate on the use case and provide a testing example using [ent/bug](https://github.com/ent/bug)?

The use case is simple. I have a lot of tests created using standard go testing features. At the beginning of each tests, there's usually a setup func call that would spin up the database server, create required databases depending on the case, and finally run migrations to put db at latest version. When running sequentially everything goes fine, but adding `t.Parallel()` to tests to allow them running in parallel, the Schema load fails with a not found package error.

I'll create the `ent/bug` testing example when I get some free time. But the relevant piece of code here is the following (I think):

```
if err := os.MkdirAll(".entc", os.ModePerm); err != nil {
		return nil, err
	}
	target := fmt.Sprintf(".entc/%s.go", filename(spec.PkgPath))
	if err := os.WriteFile(target, buf, 0644); err != nil {
		return nil, fmt.Errorf("entc/load: write file %s: %w", target, err)
	}
	defer os.RemoveAll(".entc")
	out, err := run(target)
	if err != nil {
		return nil, err
	}
```

<img width="1023" alt="image" src="https://user-images.githubusercontent.com/8674703/172233291-b1c0f3e1-8d67-4954-9bd6-e45e81116731.png">
 I think that you guys are removing the folder in which the schema gets written temporarily and that might be the cause of some tests just complaining about it when everything is running.
 
 Code from 1 of the tests is as follows:
 
<img width="906" alt="image" src="https://user-images.githubusercontent.com/8674703/172233617-a88ea3bc-5bd8-4ba9-9041-93a29c362810.png">

Basically all of our tests follow the same structure, call `t.Parallel()` to allow running multiple of the tests in the package in parallel, then we call `setup` func where schema load error occurs.
